
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../chisel/">
      
      
        <link rel="next" href="../signals/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Alleloscope - HCBench</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.84d31ad4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#alleloscope-parser" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="HCBench" class="md-header__button md-logo" aria-label="HCBench" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            HCBench
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Alleloscope
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://compbioclub.github.io/HCBench" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Compbioclub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="HCBench" class="md-nav__button md-logo" aria-label="HCBench" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    HCBench
  </label>
  
    <div class="md-nav__source">
      <a href="https://compbioclub.github.io/HCBench" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    Compbioclub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Parsers
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Parsers
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../chisel/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CHISEL
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Alleloscope
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Alleloscope
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-start" class="md-nav__link">
    <span class="md-ellipsis">
      üöÄ Quick Start
    </span>
  </a>
  
    <nav class="md-nav" aria-label="üöÄ Quick Start">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-parse-the-cna-matrix-from-rds" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ 1. Parse the CNA Matrix from RDS
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-parse-the-cluster-file" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ 2. Parse the Cluster File
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-parse-the-bin-counts-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ 3. Parse the bin counts matrix
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-parse-the-vaf-sparse-matrices" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ 4. Parse the VAF sparse matrices
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#initialization" class="md-nav__link">
    <span class="md-ellipsis">
      ‚öôÔ∏è Initialization
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#core-methods" class="md-nav__link">
    <span class="md-ellipsis">
      üß† Core Methods
    </span>
  </a>
  
    <nav class="md-nav" aria-label="üß† Core Methods">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alleloscopeparserrun" class="md-nav__link">
    <span class="md-ellipsis">
      AlleloscopeParser.run()
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alleloscopeparserget_clustercluster_file_path" class="md-nav__link">
    <span class="md-ellipsis">
      AlleloscopeParser.get_cluster(cluster_file_path)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alleloscopeparserget_bin_countscounts_file" class="md-nav__link">
    <span class="md-ellipsis">
      AlleloscopeParser.get_bin_counts(counts_file)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alleloscopeparserget_vaf_matrixvaf_file_path-output_pathnone-min_dp1-min_cells1-prefixcellsnp" class="md-nav__link">
    <span class="md-ellipsis">
      AlleloscopeParser.get_VAF_matrix(vaf_file_path, output_path=None, min_dp=1, min_cells=1, prefix="cellSNP")
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generating-a-cluster-file-in-r" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ Generating a Cluster File in R
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../signals/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SIGNALS
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../seacon/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SEACON
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cnrein/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    CNRein
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    GTbench
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            GTbench
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../gtbench/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../gtbench/clusterConsistency/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    clusterConsistency
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../gtbench/cloneSizebycluster/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    cloneSizebycluster
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../gtbench/cellprofile/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    cellprofile
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Realbench
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Realbench
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../realbench/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-start" class="md-nav__link">
    <span class="md-ellipsis">
      üöÄ Quick Start
    </span>
  </a>
  
    <nav class="md-nav" aria-label="üöÄ Quick Start">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-parse-the-cna-matrix-from-rds" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ 1. Parse the CNA Matrix from RDS
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-parse-the-cluster-file" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ 2. Parse the Cluster File
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-parse-the-bin-counts-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ 3. Parse the bin counts matrix
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-parse-the-vaf-sparse-matrices" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ 4. Parse the VAF sparse matrices
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#initialization" class="md-nav__link">
    <span class="md-ellipsis">
      ‚öôÔ∏è Initialization
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#core-methods" class="md-nav__link">
    <span class="md-ellipsis">
      üß† Core Methods
    </span>
  </a>
  
    <nav class="md-nav" aria-label="üß† Core Methods">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alleloscopeparserrun" class="md-nav__link">
    <span class="md-ellipsis">
      AlleloscopeParser.run()
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alleloscopeparserget_clustercluster_file_path" class="md-nav__link">
    <span class="md-ellipsis">
      AlleloscopeParser.get_cluster(cluster_file_path)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alleloscopeparserget_bin_countscounts_file" class="md-nav__link">
    <span class="md-ellipsis">
      AlleloscopeParser.get_bin_counts(counts_file)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alleloscopeparserget_vaf_matrixvaf_file_path-output_pathnone-min_dp1-min_cells1-prefixcellsnp" class="md-nav__link">
    <span class="md-ellipsis">
      AlleloscopeParser.get_VAF_matrix(vaf_file_path, output_path=None, min_dp=1, min_cells=1, prefix="cellSNP")
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generating-a-cluster-file-in-r" class="md-nav__link">
    <span class="md-ellipsis">
      üß¨ Generating a Cluster File in R
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="alleloscope-parser">Alleloscope Parser</h1>
<p>This module provides <code>AlleloscopeParser</code>, specialized for parsing <strong>Alleloscope</strong> outputs and exporting standardized matrices and helper files used by hcbench workflows.</p>
<p>Key features:</p>
<ul>
<li>Read Alleloscope <strong>RDS</strong> outputs (genotypes and segmentation table) and convert them into a unified haplotype level CNA matrix</li>
<li>Optional region splitting by a user defined <code>bin_size</code></li>
<li>Parse Alleloscope cluster assignments into a standardized <code>clusters.csv</code></li>
<li>Export bin-level count matrices as <code>bin_counts.csv</code></li>
<li>Convert a cellSNP-like VAF long table into sparse matrix outputs</li>
</ul>
<h2 id="quick-start">üöÄ Quick Start</h2>
<h3 id="1-parse-the-cna-matrix-from-rds">üß¨ 1. Parse the CNA Matrix from RDS</h3>
<pre><code class="language-python">from hcbench.parsers.alleloscope import AlleloscopeParser

allelo_output = &quot;/output/alleloscope/&quot;
genotypes_rds = &quot;/demo_output/alleloscope/output/rds/genotypes.rds&quot;
seg_table_rds = &quot;/demo_output/alleloscope/output/rds/seg_table.rds&quot;

alleloscope_parser = AlleloscopeParser(
    output_path=allelo_output,
    genotypes_rds_path=genotypes_rds,
    seg_table_rds_path=seg_table_rds,
    barcode_path=None,
    bin_size=100000,          # set an integer to split regions, e.g. 100000
    start_offset=1,         # shift start coordinate by +1 if needed
)

alleloscope_parser.run()
</code></pre>
<p>After running, the parser will read the two RDS files and save results to the output directory, typically containing files, for example:</p>
<pre><code>/output/alleloscope/
  haplotype_combined.csv
  haplotype_1.csv
  haplotype_2.csv
  minor.csv
  major.csv
  minor_major.csv
</code></pre>
<h3 id="2-parse-the-cluster-file">üß¨ 2. Parse the Cluster File</h3>
<pre><code class="language-python">cluster_file = &quot;/demo_output/alleloscope/output/clusters.csv&quot;
alleloscope_parser.get_cluster(cluster_file)
</code></pre>
<p>After running this command, the following standardized file will be created:</p>
<pre><code>/output/alleloscope/
  clusters.csv
</code></pre>
<p>with content:</p>
<pre><code>cell_id,clone_id
cellA,1
cellB,1
cellC,2
</code></pre>
<h3 id="3-parse-the-bin-counts-matrix">üß¨ 3. Parse the bin counts matrix</h3>
<pre><code>counts_file = &quot;/demo_output/alleloscope/input/counts/raw_counts.tsv&quot;
alleloscope_parser.get_bin_counts(counts_file)
</code></pre>
<p>After running this command, the following standardized file will be created:</p>
<pre><code>/output/alleloscope/
  bin_counts.csv
</code></pre>
<h3 id="4-parse-the-vaf-sparse-matrices">üß¨ 4. Parse the VAF sparse matrices</h3>
<p>This method assumes <code>vaf_file_path</code> is a directory containing cellSNP style outputs such as:</p>
<ul>
<li><code>cellSNP.samples.tsv</code></li>
<li><code>cellSNP.tag.AD.mtx</code></li>
<li><code>cellSNP.tag.DP.mtx</code></li>
<li><code>cellSNP.base.vcf.gz</code></li>
</ul>
<pre><code>alleloscope_parser.get_VAF_matrix(
    vaf_file_path=&quot;/demo_output/alleloscope/cellSNP_out&quot;,
    min_dp=3,
    min_cells=10,
    prefix=&quot;cellSNP&quot;,
)
</code></pre>
<p>Output:</p>
<pre><code>/output/alleloscope/
  VAF/
    cellSNP.VAF.filtered.mtx
    ...
</code></pre>
<h2 id="initialization">‚öôÔ∏è Initialization</h2>
<pre><code>AlleloscopeParser(
    output_path: str,
    genotypes_rds_path: str,
    seg_table_rds_path: str,
    barcode_path: str | None = None,
    bin_size: int | None = None,
    add_chr_prefix: bool = True,
    start_offset: int = 1,
    **kwargs
)
</code></pre>
<p><strong><code>output_path</code></strong> : base output directory.</p>
<p><strong><code>genotypes_rds_path</code></strong> : Path to the Alleloscope genotypes RDS file.</p>
<p><strong><code>seg_table_rds_path</code></strong>: Path to the Alleloscope segmentation table RDS file. It is used to map region identifiers into standardized region labels like:</p>
<ul>
<li><code>chr&lt;CHR&gt;:&lt;START+start_offset&gt;-&lt;END&gt;</code> when <code>add_chr_prefix=True</code></li>
<li><code>&lt;CHR&gt;:&lt;START+start_offset&gt;-&lt;END&gt;</code> when <code>add_chr_prefix=False</code></li>
</ul>
<p><strong><code>barcode_path</code></strong> (optional) :Path to a barcode mapping file used to remap cell IDs</p>
<p><strong><code>bin_size</code></strong> (optional) :If set, regions are split into fixed sized. </p>
<p><strong><code>add_chr_prefix</code></strong> (default <code>True</code>) :Controls whether to prepend <code>"chr"</code> when forming the region label from the segmentation table.</p>
<p><strong><code>start_offset</code></strong> (default <code>1</code>) :Adds an offset to the start coordinate when forming region labels.</p>
<h2 id="core-methods">üß† Core Methods</h2>
<h3 id="alleloscopeparserrun"><code>AlleloscopeParser.run()</code></h3>
<p>Executes the standard pipeline for the CNA matrix:</p>
<ul>
<li>Reads the provided <code>genotypes_rds_path</code> and <code>seg_table_rds_path</code>.</li>
<li>Standardizes genomic regions based on <code>start_offset</code> and <code>add_chr_prefix</code>.</li>
<li>Optionally splits regions into smaller bins if <code>bin_size</code> is provided.</li>
<li>Writes the unified haplotype matrices directly into <code>output_path</code>.</li>
</ul>
<hr />
<h3 id="alleloscopeparserget_clustercluster_file_path"><code>AlleloscopeParser.get_cluster(cluster_file_path)</code></h3>
<p>Parses an Alleloscope cluster mapping file and writes a standardized CSV.</p>
<p><strong>Input</strong></p>
<ul>
<li><code>cluster_file_path</code>: CSV file containing at least the cell identifiers and their corresponding cluster/clone IDs.</li>
</ul>
<p><strong>Output</strong> writes to:</p>
<ul>
<li><code>{self.output_path}/clusters.csv</code>, ensuring <code>cell_id</code> and <code>clone_id</code> column formatting.</li>
</ul>
<hr />
<h3 id="alleloscopeparserget_bin_countscounts_file"><code>AlleloscopeParser.get_bin_counts(counts_file)</code></h3>
<p>Creates a region-by-cell wide matrix of per-bin counts.</p>
<p><strong>Input</strong></p>
<ul>
<li><code>counts_file</code>: Path to the raw bin counts TSV file.</li>
</ul>
<p><strong>Output</strong> writes to:</p>
<ul>
<li><code>{self.output_path}/bin_counts.csv</code></li>
</ul>
<p>This is a wide matrix:</p>
<ul>
<li>rows: <code>region</code></li>
<li>columns: cells</li>
<li>values: counts</li>
</ul>
<hr />
<h3 id="alleloscopeparserget_vaf_matrixvaf_file_path-output_pathnone-min_dp1-min_cells1-prefixcellsnp"><code>AlleloscopeParser.get_VAF_matrix(vaf_file_path, output_path=None, min_dp=1, min_cells=1, prefix="cellSNP")</code></h3>
<p>Converts cellSNP-like VAF outputs into sparse matrix outputs.</p>
<p><strong>Input format</strong></p>
<p><code>vaf_file_path</code> must be a directory containing cellSNP-style outputs such as:</p>
<ul>
<li><code>cellSNP.samples.tsv</code></li>
<li><code>cellSNP.tag.AD.mtx</code></li>
<li><code>cellSNP.tag.DP.mtx</code></li>
<li><code>cellSNP.base.vcf.gz</code></li>
</ul>
<p><strong>Parameters</strong></p>
<p><code>output_path</code> (optional)</p>
<ul>
<li>If provided: outputs under <code>{output_path}/VAF</code>, else outputs under <code>{self.output_path}/VAF</code>.</li>
</ul>
<pre><code>min_dp
</code></pre>
<ul>
<li>Filter low depth sites.</li>
</ul>
<pre><code>min_cells
</code></pre>
<ul>
<li>Filter sites supported by too few cells.</li>
</ul>
<pre><code>prefix
</code></pre>
<ul>
<li>Output file prefix (default: <code>cellSNP</code>).</li>
</ul>
<p><strong>Output</strong></p>
<p>Creates a <code>VAF/</code> directory containing the sparse matrix (<code>.mtx</code>) files.</p>
<h2 id="generating-a-cluster-file-in-r">üß¨ Generating a Cluster File in R</h2>
<p>If you wish to generate the Alleloscope clustering file from your processed object in R,
 you can use the following commands:</p>
<pre><code>linplot = Lineage_plot(Obj_filtered = Obj_filtered, nSNP = 2000, nclust = 10)
write.csv(linplot, file = &quot;cluster.csv&quot;, row.names = TRUE)
</code></pre>
<p>This will create a <code>cluster.csv</code> file that can be used as input to <code>AlleloscopeParser.get_cluster()</code> in Python.</p>
<hr />












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>